python/nav/portadmin/napalm/juniper.py

```diff
@@ -123,6 +124,22 @@ def is_els(self) -> bool:
             self._is_els = self.device.device.facts.get("switch_style") == "VLAN_L2NG"
         return self._is_els
 
+    @property
+    def _els_ethernet_switching_interface_table(self):
+        return (
+            ElsEthernetSwitchingInterfaceTable
+            if self.junos_major_version < 20
+            else ElsEthernetSwitchingInterfaceTableJunOS20
+        )
+
+    @property
+    def junos_major_version(self) -> int:
+        """Returns the major JunOS version running on the switch"""
+        version = self.device.device.facts.get("version")
+        if version:
+            major, _ = version.split(".", maxsplit=1)
+            return int(major)
```

Comment:
Let's hope you never need to match on minor version too... not holding my breath